src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -900,20 +900,39 @@ void CompositionalMultiphaseBase::initializeFluidState( MeshLevel & mesh,
                                               [&]( localIndex const,
                                                    ElementSubRegionBase & subRegion )
   {
+    string const & fluidName = subRegion.template getReference< string >( viewKeyStruct::fluidNamesString() );
+    MultiFluidBase const & fluid = getConstitutiveModel< MultiFluidBase >( subRegion, fluidName );
+    string_array const & componentNames = fluid.componentNames();
+
+    // check the global component fractions values
+    arrayView2d< real64 const, compflow::USD_COMP > const compFrac =
+      subRegion.getField< flow::globalCompFraction >();
+
+    forAll< parallelDevicePolicy<> >( subRegion.size(), [&] GEOS_HOST_DEVICE ( localIndex const ei )
```

Comment:
```suggestion
    forAll< parallelDevicePolicy<> >( subRegion.size(), [&,compFracView=compFrac] GEOS_HOST_DEVICE ( localIndex const ei )
```
I think we have to capture the `arrayView` by value.

Reply:
fixed